packages/core/src/types.ts

```diff
@@ -58,6 +58,29 @@ export interface LogoutUrlResponse {
   data: { url: LogoutUrl };
 }
 
+/**
+ * A Promise to a response that can be awaited like normal,
+ * but at the risk of not getting all results due to API limits.
+ * In which case a nextCursor field is provided to request the next page.
+ * Helper methods are provided to abstract away the pagination.
+ *
+ * Example using `client.timeseries.list`:
+ * ```js
+ * const response = client.timeseries.list({ filter: { assetIds: [ASSET_ID] } });
+ * ```
+ *
+ * You can iterate through all returned items like so:
+ * ```js
+ * for await (const value of response) {
+ *   // do something to value
+ * }
+ * ```
+ *
+ * Or get an array of up to 1000 items like so:
+ * ```js
+ * const timeseries = await response.autoPagingToArray({ limit: 1000 });
```

Comment:
but this is a thing. you can get more than 1000
`client.timeseries.list({ filter: { assetIds: [ASSET_ID] }, limit: 1000 }). autoPagingToArray({ limit: 5000 });` will iterate 5 times

you can also use `autoPagingToArray({ limit: Infinity });` to fetch all